corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -212,6 +212,16 @@ def get_last_form_submissions_by_user(domain, user_ids, app_id=None, xmlns=None)
     return result
 
 
+def get_last_form_for_apps(apps, user_id):
+    forms = []
+    for app_id in apps:
+        query = FormES().app(app_id).user_id(user_id).sort('received_on', desc=True).size(1)
+        hits = query.run().hits
+        if hits:
+            forms.append(hits[0])
+    return forms
```

Comment:
Rather than doing one query per app you could to a top hits aggregation by app_id similar to this: https://github.com/dimagi/commcare-hq/blob/e17eb719eef452eed3d0dda0c7fd3f62d1f94d9b/corehq/apps/reports/analytics/esaccessors.py#L33-L49